### PR TITLE
chore(default): migrate

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

refactoring

**Summary**

- azure pipelines
- refactor test on import/export
- update deps
- move configurations to own files
- commitlint + husky + lint-staged

**Does this PR introduce a breaking change?**

No

**Other information**

No

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,11 +2,8 @@ module.exports = {
   root: true,
   plugins: ['prettier'],
   extends: ['@webpack-contrib/eslint-config-webpack'],
-  parserOptions: {
-    sourceType: 'script',
-  },
   rules: {
     'prettier/prettier': ['error'],
-    strict: ['error', 'safe'],
+    strict: ['off'],
```
Comment (evilebottnawi) [inline .eslintrc.js]: We need migrate on babel in next PR

**Comment by evilebottnawi** [comment]
npm audit works is hell :disappointed:

File: package.json [hunk 2]
```diff
@@ -12,69 +12,60 @@
...
+    "commitlint": "commitlint --from=master",
+    "lint": "eslint --cache lib test",
+    "release": "standard-version",
+    "security": "npm audit",
     "test:only": "jest",
-    "test:coverage": "jest --coverage",
-    "test": "jest",
-    "release": "standard-version"
+    "test:watch": "jest --watch",
+    "test:coverage": "jest --collectCoverageFrom='lib/**/*.js' --coverage",
```
Comment (hiroppy) [inline package.json]: I want to create `jest.config.js` like webpack-dev-server.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
+# Contributing in @webpack-contrib
```
Comment (hiroppy) [inline .github/CONTRIBUTING.md]:
```diff
- @webpack-contrib
+ webpack-dev-middleware
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+We'd always love contributions to further improve the webpack / webpack-contrib ecosystem!
```
Comment (hiroppy) [inline .github/CONTRIBUTING.md]:
```diff
- webpack-contrib
+ webpack-dev-middleware
```

**Comment by hiroppy** [comment]
@evilebottnawi Can you add renovate to this repository?

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (evilebottnawi) [reply .github/CONTRIBUTING.md]: It is ord, let's use `webpack`

File: package.json (hunk 2 above)
Comment (evilebottnawi) [reply package.json]: I think we need do same in `webpack-default` and here

**Comment by evilebottnawi** [comment]
@hiroppy What we should do here to enable this?

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (evilebottnawi) [reply .github/CONTRIBUTING.md]: It is org, let's use `webpack`

File: package.json [hunk 5]
```diff
@@ -12,69 +12,60 @@
...
+    "commitlint": "commitlint --from=master",
+    "lint": "eslint --cache lib test",
+    "release": "standard-version",
+    "security": "npm audit",
     "test:only": "jest",
-    "test:coverage": "jest --coverage",
-    "test": "jest",
-    "release": "standard-version"
+    "test:coverage": "npm run test:only -- --coverage",
+    "test:watch": "npm run test:only --watch",
```
Comment (evilebottnawi) [inline package.json]: /cc @hiroppy watch should be without coverage, so better use `test:only` (we need fix it in webpack-dev-server)

Comment (hiroppy) [reply package.json]: > watch should be without coverage

hm... honestly, I want to be able to see coverage when running as `test:watch`.

**Pull request merged**